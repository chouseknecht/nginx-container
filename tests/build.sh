#!/bin/bash

ansible-playbook build.yml | tee build.out
grep failed=0 build.out | grep unreachable=0 >/dev/null

if [ "$?" == "0" ]; then
    # if all looks good, give it a go
    echo "Build completed successfully. Starting the project..."
    ansible-container run
 fi 
