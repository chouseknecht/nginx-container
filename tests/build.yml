- name: Build the project  
  hosts: localhost
  connection: local
  gather_facts: no
  tasks:

  - name: Remove the ansible directory
    file:
      path: ./ansible
      state: absent

  - name: Init the ansible directory 
    command: ansible-container init
    
  - name: Install this role with latest commit   
    shell: "ansible-container install git+https://github.com/chouseknecht/nginx-container.git,$(git rev-parse HEAD)"
      
  - name: Build the project
    command: ansible-container build
    register: output

  - name: Show the build output
    debug:
      var: output.stdout_lines

