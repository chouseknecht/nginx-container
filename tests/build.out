
PLAY [Build the project] *******************************************************

TASK [Remove the ansible directory] ********************************************
changed: [localhost]

TASK [Init the ansible directory] **********************************************
changed: [localhost]

TASK [Install this role with latest commit] ************************************
changed: [localhost]

TASK [Build the project] *******************************************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["ansible-container", "build"], "delta": "0:00:30.176767", "end": "2016-10-31 17:36:35.082090", "failed": true, "rc": 1, "start": "2016-10-31 17:36:04.905323", "stderr": "Starting Docker Compose engine to build your images...\nCleaning up Ansible Container builder...\nStopping ansible_nginx_1 ... \r\n\u001b[1A\u001b[2K\rStopping ansible_nginx_1 ... done\r\u001b[1BAnsible playbook run failed.\nCleaning up Ansible Container builder...\nAnsible build failed", "stdout": "Attaching to ansible_ansible-container_1\nAttaching to ansible_ansible-container_1, ansible_nginx_1\n\u001b[36mansible-container_1  |\u001b[0m - extracting nginx-container to /etc/ansible/roles/nginx-container\n\u001b[36mansible-container_1  |\u001b[0m - nginx-container was installed successfully\n\u001b[36mansible-container_1  |\u001b[0m Host nginx running\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m PLAY [all] *********************************************************************\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m PLAY [nginx] *******************************************************************\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [setup] *******************************************************************\n\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Enable nginx repo] *************************************\n\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Intsall nginx] *****************************************\n\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Create nginx user] *************************************\n\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Set permissions on /var/log/nginx and /var/www/nginx] **\n\u001b[36mansible-container_1  |\u001b[0m ok: [nginx] => (item=/var/log/nginx)\n\u001b[36mansible-container_1  |\u001b[0m changed: [nginx] => (item=/var/www/nginx)\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Remove pid file, if it exists] *************************\n\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Set permissions on pid file directory] *****************\n\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Remove /var/cache/nginx, if it exists] *****************\n\u001b[36mansible-container_1  |\u001b[0m changed: [nginx]\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Create /var/cache/nginx] *******************************\n\u001b[36mansible-container_1  |\u001b[0m changed: [nginx]\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Get rid of the defaults dropped into /etc/nginx/conf.d] \n\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Link nginx log system to stderr and stdout] ************\n\u001b[36mansible-container_1  |\u001b[0m ok: [nginx] => (item={u'path': u'/var/log/nginx/access.log', u'src': u'/dev/stdout'})\n\u001b[36mansible-container_1  |\u001b[0m ok: [nginx] => (item={u'path': u'/var/log/nginx/error.log', u'src': u'/dev/stderr'})\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Create default nginx configuration] ********************\n\u001b[36mansible-container_1  |\u001b[0m changed: [nginx]\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Copy default content to /var/www/nginx] ****************\n\u001b[36mansible-container_1  |\u001b[0m fatal: [nginx]: FAILED! => {\"changed\": false, \"failed\": true, \"msg\": \"Remote copy does not support recursive copy of directory: /usr/share/nginx/html\"}\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m NO MORE HOSTS LEFT *************************************************************\n\u001b[36mansible-container_1  |\u001b[0m  [WARNING]: Could not create retry file 'main.retry'.         [Errno 2] No such\n\u001b[36mansible-container_1  |\u001b[0m file or directory: ''\n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible-container_1  |\u001b[0m PLAY RECAP *********************************************************************\n\u001b[36mansible-container_1  |\u001b[0m nginx                      : ok=12   changed=4    unreachable=0    failed=1   \n\u001b[36mansible-container_1  |\u001b[0m \n\u001b[36mansible_ansible-container_1 exited with code 1\n\u001b[0mAborting on container exit...", "stdout_lines": ["Attaching to ansible_ansible-container_1", "Attaching to ansible_ansible-container_1, ansible_nginx_1", "\u001b[36mansible-container_1  |\u001b[0m - extracting nginx-container to /etc/ansible/roles/nginx-container", "\u001b[36mansible-container_1  |\u001b[0m - nginx-container was installed successfully", "\u001b[36mansible-container_1  |\u001b[0m Host nginx running", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m PLAY [all] *********************************************************************", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m PLAY [nginx] *******************************************************************", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [setup] *******************************************************************", "\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Enable nginx repo] *************************************", "\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Intsall nginx] *****************************************", "\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Create nginx user] *************************************", "\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Set permissions on /var/log/nginx and /var/www/nginx] **", "\u001b[36mansible-container_1  |\u001b[0m ok: [nginx] => (item=/var/log/nginx)", "\u001b[36mansible-container_1  |\u001b[0m changed: [nginx] => (item=/var/www/nginx)", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Remove pid file, if it exists] *************************", "\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Set permissions on pid file directory] *****************", "\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Remove /var/cache/nginx, if it exists] *****************", "\u001b[36mansible-container_1  |\u001b[0m changed: [nginx]", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Create /var/cache/nginx] *******************************", "\u001b[36mansible-container_1  |\u001b[0m changed: [nginx]", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Get rid of the defaults dropped into /etc/nginx/conf.d] ", "\u001b[36mansible-container_1  |\u001b[0m ok: [nginx]", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Link nginx log system to stderr and stdout] ************", "\u001b[36mansible-container_1  |\u001b[0m ok: [nginx] => (item={u'path': u'/var/log/nginx/access.log', u'src': u'/dev/stdout'})", "\u001b[36mansible-container_1  |\u001b[0m ok: [nginx] => (item={u'path': u'/var/log/nginx/error.log', u'src': u'/dev/stderr'})", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Create default nginx configuration] ********************", "\u001b[36mansible-container_1  |\u001b[0m changed: [nginx]", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m TASK [nginx-container : Copy default content to /var/www/nginx] ****************", "\u001b[36mansible-container_1  |\u001b[0m fatal: [nginx]: FAILED! => {\"changed\": false, \"failed\": true, \"msg\": \"Remote copy does not support recursive copy of directory: /usr/share/nginx/html\"}", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m NO MORE HOSTS LEFT *************************************************************", "\u001b[36mansible-container_1  |\u001b[0m  [WARNING]: Could not create retry file 'main.retry'.         [Errno 2] No such", "\u001b[36mansible-container_1  |\u001b[0m file or directory: ''", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible-container_1  |\u001b[0m PLAY RECAP *********************************************************************", "\u001b[36mansible-container_1  |\u001b[0m nginx                      : ok=12   changed=4    unreachable=0    failed=1   ", "\u001b[36mansible-container_1  |\u001b[0m ", "\u001b[36mansible_ansible-container_1 exited with code 1", "\u001b[0mAborting on container exit..."], "warnings": []}
	to retry, use: --limit @/Users/chouseknecht/projects/nginx-container/tests/build.retry

PLAY RECAP *********************************************************************
localhost                  : ok=3    changed=3    unreachable=0    failed=1   

